src/platform/nrfconnect/wifi/NrfWiFiDriver.cpp

```diff
@@ -24,51 +26,107 @@
 
 using namespace ::chip;
 using namespace ::chip::DeviceLayer::Internal;
+using namespace ::chip::DeviceLayer::PersistedStorage;
 
 namespace chip {
 namespace DeviceLayer {
 namespace NetworkCommissioning {
 
 CHIP_ERROR NrfWiFiDriver::Init(NetworkStatusChangeCallback * networkStatusChangeCallback)
 {
+    LoadFromStorage();
+
+    if (mStagingNetwork.IsConfigured())
+    {
+        // TODO: Make WiFiManager able to change networks and disconnect
+        ReturnErrorOnFailure(WiFiManager::Instance().AddNetwork(mStagingNetwork.GetSsid(), mStagingNetwork.GetPass()));
+        ReturnErrorOnFailure(WiFiManager::Instance().Connect());
+    }
+
     return CHIP_NO_ERROR;
 }
 
 void NrfWiFiDriver::Shutdown() {}
 
 CHIP_ERROR NrfWiFiDriver::CommitConfiguration()
 {
+    ReturnErrorOnFailure(KeyValueStoreMgr().Put(kPassKey, mStagingNetwork.pass, mStagingNetwork.passLen));
+    ReturnErrorOnFailure(KeyValueStoreMgr().Put(kSsidKey, mStagingNetwork.ssid, mStagingNetwork.ssidLen));
+
     return CHIP_NO_ERROR;
 }
 
 CHIP_ERROR NrfWiFiDriver::RevertConfiguration()
 {
+    LoadFromStorage();
+
+    if (mStagingNetwork.IsConfigured())
+    {
+        ReturnErrorOnFailure(WiFiManager::Instance().AddNetwork(mStagingNetwork.GetSsid(), mStagingNetwork.GetPass()));
+        ReturnErrorOnFailure(WiFiManager::Instance().Connect());
+    }
+
     return CHIP_NO_ERROR;
 }
 
 Status NrfWiFiDriver::AddOrUpdateNetwork(ByteSpan ssid, ByteSpan credentials, MutableCharSpan & outDebugText,
                                          uint8_t & outNetworkIndex)
 {
-    WiFiManager::Instance().AddNetwork(ssid, credentials);
+    outDebugText    = {};
+    outNetworkIndex = 0;
+
+    VerifyOrReturnError(!mStagingNetwork.IsConfigured() || ssid.data_equal(mStagingNetwork.GetSsid()), Status::kBoundsExceeded);
```

Comment:
Because if the ssid doesn't match the current one, it means the command is trying to add another network and the maximum number of network supported by this implementation is 1.

Reply:
Ok, got it.